crates/packager/src/package/pacman/mod.rs

```diff
@@ -0,0 +1,145 @@
+    let pkg_path = config.out_dir().join(&package_name);
+    let pkgbuild_path = pkg_path.with_file_name("PKGBUILD");
+
+    tracing::info!("Packaging {} ({})", package_name, pkg_path.display());
+
+    tracing::debug!("Generating data");
+    let _ = generate_data(config, &pkg_dir)?;
+
+    tracing::debug!("Copying files specified in `deb.files`");
+    if let Some(files) = config.deb().and_then(|d| d.files.as_ref()) {
+        copy_custom_files(files, &pkg_dir)?;
+    }
+
+    // Apply tar/gzip to create the final package file.
+    tracing::debug!("Creating package archive using tar and gzip");
+    let data_tar_gz_path = tar_and_gzip_dir(pkg_dir)?;
+    std::fs::copy(data_tar_gz_path, &pkg_path)?;
+
+    tracing::info!("Generating PKGBUILD: {}", pkgbuild_path.display());
+    generate_pkgbuild_file(config, arch, pkgbuild_path.as_path(), pkg_path.as_path())?;
+
+    Ok(vec![pkg_path])
+}
+
+/// Generates the pacman PKGBUILD file.
+/// For more information about the format of this file, see
+/// <https://wiki.archlinux.org/title/PKGBUILD>
+fn generate_pkgbuild_file(
+    config: &Config,
+    arch: &str,
+    dest_dir: &Path,
+    package_path: &Path,
+) -> crate::Result<()> {
+    let pkgbuild_path = dest_dir.with_file_name("PKGBUILD");
+    let mut file = util::create_file(&pkgbuild_path)?;
+
+    if let Some(authors) = &config.authors {
+        writeln!(file, "# Maintainer: {}", authors.join(", "))?;
+    }
+    writeln!(file, "pkgname={}-bin", AsKebabCase(&config.product_name))?;
+    writeln!(file, "pkgver={}", config.version)?;
+    writeln!(file, "pkgrel=1")?;
+    writeln!(file, "epoch=")?;
+    writeln!(
+        file,
+        "pkgdesc=\"{}\"",
+        config.description.as_deref().unwrap_or("")
+    )?;
+    writeln!(file, "arch=('{}')", arch)?;
+
+    if let Some(homepage) = &config.homepage {
+        writeln!(file, "url=\"{}\"", homepage)?;
+    }
+
+    let dependencies = config
+        .pacman()
+        .cloned()
+        .and_then(|d| d.depends)
+        .unwrap_or_default();
+    writeln!(file, "depends=({})", dependencies.join(" \n"))?;
+
+    let provides = config
+        .pacman()
+        .cloned()
+        .and_then(|d| d.provides)
+        .unwrap_or_default();
+    writeln!(file, "provides=({})", provides.join(" \n"))?;
+
+    let conflicts = config
+        .pacman()
+        .cloned()
+        .and_then(|d| d.conflicts)
+        .unwrap_or_default();
+    writeln!(file, "conflicts=({})", conflicts.join(" \n"))?;
+
+    let replaces = config
+        .pacman()
+        .cloned()
+        .and_then(|d| d.replaces)
+        .unwrap_or_default();
+    writeln!(file, "replaces=({})", replaces.join(" \n"))?;
+
+    writeln!(file, "options=(!lto)")?;
+    let source = config
+        .pacman()
+        .cloned()
+        .and_then(|d| d.source)
+        .unwrap_or_default();
+
+    if source.is_empty() {
+        writeln!(file, "source=({:?})", package_path.file_name().unwrap())?;
```

Comment:
do we need to set this if not provided by user?

Reply:
Yes, because it will make it more estimable for the user on what link to replace by source.

It is not necessary though